### DISCUSSION TITLE
Extending simple-commerce

### DISCUSSION THREAD

**Discussion by jhgeluk** [duncanmcclean/simple-commerce#852] [Help]
Hello,

I'm quite new to Statamic, but I was wondering if it's possible to fork/extend simple-commerce. I'm building a website where people can receive a quote for multiple services (products) at once. Therefore I don't need coupons, but I do need the cart, etc.

I think this is would be a good foundation to start with. What would be the best practice to extend/fork this addon?

**Comment by duncanmcclean** [comment]
You shouldn't need to fork Simple Commerce at all to achieve that - everything you need to do should be achievable in your app, by building on top of what Simple Commerce offers you.

For your situation, where you need the cart but nothing else. What are your requirements in terms of needing to build on top of Simple Commerce?

**Comment by jhgeluk** [comment]
You could see it as an "e-commerce" website but instead of products, we want to make quotations.

We offer services like:
- Painting walls
- Painting stairs
- Pouring concrete

However, we need different types of information for each service, to make a quotation.

For example:
- For the painting service we need to know the amount of square meters.
- For the concrete service we need to know the amount of cubic meters.

We would like visitors to add these services to their "cart", and when going to checkout we want them to fill in the required information per service.

Hope this makes sense.

**Reply by duncanmcclean** [reply]
Regarding entering the different information, you could either:

1. Move those fields to the "add to cart" form and use the [Dynamic Pricing](https://simple-commerce.duncanmcclean.com/kb-articles/dynamic-pricing) feature to calculate the price based on what the user inputs. 

2. Have fields somewhere on the cart page asking the user to enter that information, then once that's been saved you can calculate the cost using the [Dynamic Pricing](https://simple-commerce.duncanmcclean.com/kb-articles/dynamic-pricing) feature based on what's been entered. 

Would that work?

**Reply by jhgeluk** [reply]
Alright thanks, what would be best practice to show dynamic input fields (for example: square meters, or number of walls.) per product?

**Reply by duncanmcclean** [reply]
Personally, I'd maybe have a set of checkboxes on the product entries of possible inputs you might need to show. Then, per product you can tick what you need, then show the right fields based on that.

However, I'm sure there are other ways to handle it.
